express-server/middleware/errorHandler.js

```diff
@@ -0,0 +1,19 @@
+const { ServerError } = require('../errors/ServerError');
+const { logger } = require('../logger/logger.js');
+const { HTTPStatuses } = require('../constants/HTTPStatus.js');
+
+exports.wrapError = (fn) => (req, res, next) => {
+    Promise.resolve(fn(req, res, next)).catch(next)
```

Comment:
plz explain what you're doing here

Reply:
i wrap sync and async functions with a promise, if an error is thrown from a function i catch it and then pass it to the error middleware

Reply:
it's weird. why not wrap it with try catch?